src/aiida_workgraph/engine/workgraph.py

```diff
@@ -139,10 +146,18 @@ def save_instance_state(
     def load_instance_state(
         self, saved_state: t.Dict[str, t.Any], load_context: t.Any
     ) -> None:
+        from aiida.orm.utils.log import create_logger_adapter
+
         super().load_instance_state(saved_state, load_context)
         # Load the context
         self._context = saved_state[self._CONTEXT]
-        self.set_logger(self.node.logger)
+        # TODO: avoid hardcoding the logger
+        self.node._logger = logging.getLogger(
+            "aiida.orm.nodes.process.workflow.workchain.WorkChainNode"
+        )
+        # First time the property is called after the node is stored, create the logger adapter
+        self.node._logger_adapter = create_logger_adapter(self.node._logger, self.node)
+        self.set_logger(self.node._logger_adapter)
```

Comment:
The assignment to the internal member variables looks hacky. Following aiida it looks like you just need to define the class variable `_logger` in `WorkGraphNode`. See inheritance
https://github.com/aiidateam/aiida-core/blob/main/src/aiida/orm/nodes/process/process.py#L260-L261
https://github.com/aiidateam/aiida-core/blob/f4c55f5f78cd7fde9a5b4a4e48cad2159fd666b2/src/aiida/orm/nodes/node.py#L140
https://github.com/aiidateam/aiida-core/blob/f4c55f5f78cd7fde9a5b4a4e48cad2159fd666b2/src/aiida/orm/utils/node.py#L157

So WorkGraphNode
```python
        _logger = logging.getLogger(
            "aiida.orm.nodes.process.workflow.workchain.WorkChainNode"
        )
```

and here
```suggestion
self.set_logger(self.node.logger)
```
The logger adapter is automatically created in `ProcessNode.get_logger` if not existent (`WorkGraphNode` inherits from `ProcessNode`)

Reply:
Thanks for investigating this. I think I tried this solution, but it does not work. There are still something I missed. For the moment, I hardcoded the `logger`, and will open an issue after this PR, so we can fix it in the future.